flow/connectors/mysql/cdc.go

```diff
@@ -770,3 +772,112 @@ func (c *MySqlConnector) processAlterTableQuery(ctx context.Context, catalogPool
 func posToOffsetText(pos mysql.Position) string {
 	return fmt.Sprintf("!f:%s,%x", pos.Name, pos.Pos)
 }
+
+// processTableMapEventSchema compares the TABLE_MAP_EVENT schema against the cached schema
+// and returns a TableSchemaDelta if new columns are detected (e.g., after gh-ost migration).
+// It also returns a slice mapping binlog column index to FieldDescription for efficient row processing.
+func (c *MySqlConnector) processTableMapEventSchema(
+	ctx context.Context,
+	catalogPool shared.CatalogPool,
+	req *model.PullRecordsRequest[model.RecordItems],
+	tableMap *replication.TableMapEvent,
+	sourceTableName string,
+	destinationTableName string,
+	schema *protos.TableSchema,
+	exclusion map[string]struct{},
+) ([]*protos.FieldDescription, error) {
+	newFds := make([]*protos.FieldDescription, len(tableMap.ColumnName))
+
+	// Build a set of existing column names for quick lookup
+	existingCols := make(map[string]*protos.FieldDescription, len(schema.Columns))
+	for _, col := range schema.Columns {
```

Comment:
nit: could we shave off this for-loop because this would run per event and could be expensive for wide tables? 

What i am thinking is passing in a map and conditionally add an entry for the current table if it's not in the map, or rebuild it if we detected new columns.